platform/android/src/style/sources/source.cpp

```diff
@@ -119,11 +119,12 @@ namespace android {
         ownedSource = map.getStyle().removeSource(source.getID());
 
         // The source may not be removed if any layers still reference it
-        if (!ownedSource) {
-            return;
-        }
+        return ownedSource != nullptr;
+    }
 
+    void Source::releaseJavaPeer() {
```

Comment:
Since you are separating this method from `removeFromMap`, it would be good to add back the checks for valid `ownedSource` so that it doesn't crash in the apps if called incorrectly.